2996_reflection/reflection.md

```diff
@@ -3386,14 +3393,17 @@ Add a new paragraph between [expr.eq]{.sref}/5 and /6:
-* [*.#]{.pnum} Otherwise, if one operand is a reflection of a namespace alias, alias template, or type alias and the other operand is not a reflection of the same kind of alias, they compare unequal. [A reflection of a type and a reflection of an alias to that same type do not compare equal.]{.note}
-* [*.#]{.pnum} Otherwise, if both operands are reflections of a namespace alias, alias template, or type alias, then they compare equal if their reflected aliases share the same name, are declared within the same enclosing scope, and alias the same underlying entity.
-* [*.#]{.pnum} Otherwise, if both operands are reflections of a base class specifier, then they compare equal if they are reflections of the same base class specifier.
-* [*.#]{.pnum} Otherwise, if one operand is a reflection of a base class specifier and the other is not, then they compare unequal.
-* [*.#]{.pnum} Otherwise, if neither operand is a reflection of a value, then they compare equal if they are reflections of the same entity.
-* [*.#]{.pnum} Otherwise, if one operand is a reflection of a value and the other is not, then they compare unequal.
-* [*.#]{.pnum} Otherwise, if both operands are reflections of values, then they compare equal if and only if the reflected values are _template-argument-equivalent_ ([temp.type]{.sref}).
-* [*.#]{.pnum} Otherwise the result is unspecified. [For example, if the operands are results of calls to `data_member_spec`]{.note}
+* [*.#]{.pnum} Otherwise, if both operands represent entities, then the compare equal if and only if they are reflections of the same entity.
+* [*.#]{.pnum} Otherwise, if one operand represents an entity, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if both operands represent variables, then the compare equal if and only if they are reflections of the same entity.
+* [*.#]{.pnum} Otherwise, if one operand represents a variable, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if one operand represents a namespace alias or type alias and the other operand does not represent the same kind of alias, they compare unequal. [A reflection of a type and a reflection of an alias to that same type do not compare equal.]{.note}
+* [*.#]{.pnum} Otherwise, if both operands represent namespace aliases or type aliases, then they compare equal if and only if their represented aliases share the same name, are declared within the same enclosing scope, and alias the same underlying entity.
+* [*.#]{.pnum} Otherwise, if both operands represent base class specifiers, then they compare equal if and only if they represent the same base class specifier.
+* [*.#]{.pnum} Otherwise, if one operand represents a base class specifier, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if both operands represent values, then they compare equal if and only if the reflected values are _template-argument-equivalent_ ([temp.type]{.sref}) and share the same type.
```

Comment:
template-argument-equivalent already implies same type:
```suggestion
* [*.#]{.pnum} Otherwise, if both operands represent values, then they compare equal if and only if the reflected values are _template-argument-equivalent_ ([temp.type]{.sref}).
```

Reply:
Not quite:

> they are of integral type and their values are the same, or
[(2.2)](https://eel.is/c++draft/temp.type#2.2)
they are of floating-point type and their values are identical, or
[(2.3)](https://eel.is/c++draft/temp.type#2.3)
they are of type std​::​nullptr_t, or
[(2.4)](https://eel.is/c++draft/temp.type#2.4)
they are of enumeration type and their values are the same

The values computed from the literals `3` and `3l` are considered "template-argument-equivalent" ([godbolt](https://godbolt.org/z/Y3s8hYavM)), but I _think_ we want to distinguish them here.

Reply:
Yes quite:

> Two values are [template-argument-equivalent](http://eel.is/c++draft/temp#def:template-argument-equivalent) if **they are of the same type and** [...]

Reply:
Hahahaha whoooops

Reply:
Done, thanks.